plugins/kubernetes/app/controllers/kubernetes/stages_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Kubernetes::StagesController < ResourceController
+  include CurrentProject
+  include CurrentStage
+
+  DEFAULT_BRANCH = "master"
+
+  before_action :authorize_resource!
+
+  def manifest_preview
+    # find ref and sha ... sha takes priority since it's most accurate
+    git_sha = params[:git_sha]
+    git_ref = params[:git_ref] || git_sha || current_project.release_branch || DEFAULT_BRANCH
+    git_sha ||= current_project.repository.commit_from_ref(git_ref)
+    if git_sha.empty?
+      return render body: '# Git reference not found', content_type: 'application/yaml'
```

Comment:
status 400 ?

Reply:
... if this is for js then maybe ignore the 400 ... otherwise add a comment that it's about js not failing